### PR TITLE
*: let TestEvictLeaderScheduler run in two modes

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref #8619

### What is changed and how does it work?

Previously, removeStore only takes effect when StoreIDWithRanges has the value. This PR fixes it.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/evict_leader.go [hunk 1]
```diff
@@ -182,6 +175,12 @@ func (conf *evictLeaderSchedulerConfig) pauseLeaderTransferIfStoreNotExist(id ui
 	return true, nil
 }
 
+func (conf *evictLeaderSchedulerConfig) resumeLeaderTransferIfExist(id uint64) {
```
Comment (lhy1024) [inline pkg/schedule/schedulers/evict_leader.go:178]: Do we need to cherry pick it?

**Review by lhy1024 — APPROVED** [review]

File: pkg/schedule/schedulers/evict_leader.go (hunk 1 above)
Comment (rleungx) [reply pkg/schedule/schedulers/evict_leader.go:178]: No, only master has the issue.

**Review by okJiang — APPROVED** [review]

**Comment by wuhuizuo** [comment]
> @wuhuizuo: The label(s) `?` cannot be applied. These labels are supported: `Hacktoberfest, challenge-program, ci-unstable, compatibility-breaker, high-performance, hptc, needs-cherry-pick-release-5.4, needs-cherry-pick-release-6.1, needs-cherry-pick-release-6.5, needs-cherry-pick-release-7.1, needs-cherry-pick-release-7.5, needs-cherry-pick-release-8.1, release-note, require-LGT1, wontfix, affects-5.4, affects-6.1, affects-6.5, affects-7.1, affects-7.5, affects-8.1, affects-8.3, may-affects-5.4, may-affects-6.1, may-affects-6.5, may-affects-7.1, may-affects-7.5, may-affects-8.1`.
> 
> In response to [this](https://github.com/tikv/pd/pull/8663#issuecomment-2370696110):
> 
> > /label ?
> 
> Instructions for interacting with me using PR comments are available [here](https://prow.tidb.net/command-help). If you have questions or suggestions related to my behavior, please file an issue against the [ti-community-infra/tichi](https://github.com/ti-community-infra/tichi/issues/new?title=Prow%20issue:) repository.

/remove-label needs-cherry-pick-release-5.4

**Pull request merged**